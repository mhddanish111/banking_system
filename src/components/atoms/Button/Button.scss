@import '../../../styles/colors.scss';
@import '../../../styles/boarder.scss';
@import '../../../styles/spacing.scss';
@import '../../../styles/mixins.scss';
@import '../../../styles/size.scss';
@import '../../../styles/fonts.scss';
button {
    @include font-body;
    border: none;
    background-color: $color-link;
    border-radius: $border-radius;
    color: $color-font-light;
    height: $height-medium;
    padding: 0 $space-l;
    text-transform: uppercase;
    line-height: $line-height-button;
    &.active {
      outline: none;
    }
    &:hover {
      background-color: $color-link-hover;
      transition: $hover-transition;
      box-shadow: none;
      cursor: pointer;
    }
    &:disabled {
      background-color: $color-disabled;
      cursor: default;
    }
    &.dark {
      background: transparent;
      border: 1px solid $color-white;
      &:hover {
        background-color: $color-white;
        color: $color-font-medium;
        transition: $hover-transition;
      }
    }
  }