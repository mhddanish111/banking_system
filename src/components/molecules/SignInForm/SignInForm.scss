@import "../../../styles/spacing.scss";
@import '../../../styles/boarder.scss';
@import "../../../styles/colors.scss";
.sign-in-form {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  padding: 20px;

  input {
    width: 100%;
    border-radius: 4px;
    border: $border-medium;
    padding: 10px 15px;
    margin-bottom: 10px;
    font-size: 14px;
    background-color: $color-white;
  }
  button {
    width: 100%;
  }

  .form-input {
    display: flex;
    flex-direction: column;
    margin-bottom: $space-m;
  }
}
