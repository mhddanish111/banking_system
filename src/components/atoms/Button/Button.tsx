import React from 'react';
import './Button.scss';

type buttonProps = {
  className?: string;
  children?: React.ReactNode | String;
  type?: 'submit' | 'reset' | 'button';
  ariaLabel?: string;
  disabled?: boolean;
  others?: any;
  onClick?: any;
};

const Button: React.FC<buttonProps> = ({
  className,
  children,
  type,
  disabled,
  ariaLabel,
  ...others
}) => (
  <button
    aria-label={ariaLabel}
    className={className}
    disabled={disabled}
    type={type}
    {...others}
  >
    {children}
  </button>
);

export default Button;
