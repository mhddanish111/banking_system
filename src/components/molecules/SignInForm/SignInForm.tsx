import React from 'react';
import { Link } from 'react-router-dom';
import { useForm } from 'react-hook-form';
import Button from '../../atoms/Button';
import { SignInProps, SignFormData } from '../../../type/UserType';
import './SignInForm.scss';

const SignInForm: React.FC<SignInProps> = ({ SignInHandler }) => {
  const {
    register,
    handleSubmit,
    formState: { errors },
  } = useForm<SignFormData>();
  const onSubmit = (data: SignFormData, e: any) => {
    SignInHandler(data, e);
  };
  return (
    <form
      name="sign-in-form"
      className="sign-in-form"
      onSubmit={handleSubmit(onSubmit)}
      autoComplete="off"
      aria-label="sign in form"
    >
      <div className="form-input">
        <label htmlFor="userId" aria-label="user id">
          User ID
        </label>
        <input
          type="text"
          id="userId"
          autoComplete="off"
          {...register('user', { required: 'user id is required' })}
        />
        {errors.user && <p style={{ color: 'red' }}>{errors.user.message}</p>}
      </div>
      <div className="form-input">
        <label htmlFor="password" aria-label="password">
          Password
        </label>
        <input
          type="password"
          id="password"
          autoComplete="off"
          {...register('password', { required: 'password id is required' })}
        />
        {errors.password && (
          <p style={{ color: 'red' }}>{errors.password.message}</p>
        )}
      </div>
      <div className="form-action">
        <Button type="submit">Sign In</Button>
      </div>
      <div>
        <Link to="/sign-up">Create Account</Link>
      </div>
    </form>
  );
};

export default SignInForm;
