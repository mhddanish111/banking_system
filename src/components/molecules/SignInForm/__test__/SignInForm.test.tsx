import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Button from '../../../atoms/Button';
import SignInForm from '../SignInForm';

describe('<SignInForm />', () => {
  const SignInHandler = jest.fn();
  const wrapper = shallow(<SignInForm SignInHandler={SignInHandler} />);
  it('should render the SignInForm Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should call form event', () => {
    const onSubmit = jest.fn();
    const handleSubmit = jest.fn();
    const event = { preventDefault: jest.fn() };
    const data = {
      user: 'test',
      password: 'test@123',
    };
    wrapper
      .find('.sign-in-form')
      .simulate('submit', handleSubmit(onSubmit(data, event)));
    expect(onSubmit).toHaveBeenCalled();
  });
  it('should call button event', () => {
    const onClick = jest.fn();
    const event = { preventDefault: jest.fn() };
    wrapper.find(Button).simulate('click', onClick(event));
    expect(onClick).toHaveBeenCalled();
  });
});
