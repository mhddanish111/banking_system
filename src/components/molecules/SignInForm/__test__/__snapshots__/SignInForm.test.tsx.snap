// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignInForm /> should render the SignInForm Component correctly 1`] = `
<form
  aria-label="sign in form"
  autoComplete="off"
  className="sign-in-form"
  name="sign-in-form"
  onSubmit={[Function]}
>
  <div
    className="form-input"
  >
    <label
      aria-label="user id"
      htmlFor="userId"
    >
      User ID
    </label>
    <input
      autoComplete="off"
      id="userId"
      name="user"
      onBlur={[Function]}
      onChange={[Function]}
      type="text"
    />
  </div>
  <div
    className="form-input"
  >
    <label
      aria-label="password"
      htmlFor="password"
    >
      Password
    </label>
    <input
      autoComplete="off"
      id="password"
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="form-action"
  >
    <Button
      type="submit"
    >
      Sign In
    </Button>
  </div>
  <div>
    <Link
      to="/sign-up"
    >
      Create Account
    </Link>
  </div>
</form>
`;
